testsuite/compat/src/test/java/org/jboss/as/test/compat/jpa/hibernate/HibernateJarsInDeploymentTestCase.java

```diff
@@ -79,7 +79,7 @@ private static void addHibernate42xJars(EnterpriseArchive ear) {
         lib.addClasses(SFSB1.class, HibernateJarsInDeploymentTestCase.class);
         lib.addAsManifestResource(HibernateJarsInDeploymentTestCase.class.getPackage(), "persistence.xml", "persistence.xml");
         ear.addAsManifestResource(HibernateJarsInDeploymentTestCase.class.getPackage(), "permissions.xml", "permissions.xml");
-        ear.addAsManifestResource(new StringAsset("Dependencies: org.jboss.jandex\n"), "MANIFEST.MF");
+        ear.addAsManifestResource(new StringAsset("Dependencies: org.javassist export, org.jboss.jandex\n"), "MANIFEST.MF");
```

Comment:
This test includes Hibernate jars with the deployment and verifies that the deployment copy of Hibernate is used (e.g. checks that Hibernate classloader == app classloader ).  This change is an example of what I meant about requiring applications that embed their own Hibernate jars, to also include the correct Javassist version on the application classpath, which is what the one line change does here.  This particular test is using the same version of Hibernate that the app server uses, so the app server Javassist version can be used.